libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod error;
+mod scar;
+mod submit;
+
+use sawtooth_sdk::messages::batch::BatchList;
+
+pub use error::Error;
+pub use scar::{SmartContract, SmartContractMetadata};
+use submit::{submit_batches, wait_for_batches};
+
+/// A client that can be used to submit transactions to a scabbard service.
+pub struct ScabbardClient {
+    url: String,
+}
+
+impl ScabbardClient {
+    /// Create a new `ScabbardClient` with the given `url`.
+    pub fn new(url: &str) -> Self {
```

Comment:
This should be base URL and circuit/service should be provided in submit()

Reply:
Updated